### PR TITLE
Allow for .package(url: String, revision: String) syntax

### PR DESCRIPTION
**Pull request by miggs597**
Just adding the convenience init .package(url: String, revision: String)

### CODE REVIEW

**Review by neonichu — APPROVED** [review]

**Comment by tomerd** [comment]
@swift-ci please smoke test

**Comment by miggs597** [comment]
@swift-ci please smoke test

**Comment by miggs597** [comment]
@swift-ci please smoke test

**Comment by miggs597** [comment]
@swift-ci please smoke test

File: Documentation/PackageDescription.md [hunk 1]
```diff
@@ -306,6 +306,16 @@ static func package(url: String, _ requirement: Package.Dependency.Requirement)
...
 static func package(name: String? = nil, url: String, branch: String) -> Package.Dependency
 
+/// Adds a remote package dependency given a branch requirement.
+///
+///    .package(url: "https://example.com/example-package.git", revision: "aa681bd6c61e22df0fd808044a886fc4a7ed3a65"),
+///
+/// - Parameters:
+///     - name: The name of the package, or nil to deduce it from the URL.
+///     - url: The valid Git URL of the package.
+///     - revision: A dependency requirement. See static methods on `Package.Dependency.Requirement` for available options.
```
Comment (abertelrud) [inline Documentation/PackageDescription.md:316]: This doc comment probably needs to be updated.  I think the rev can be a hash or a tag.

File: Sources/PackageDescription/PackageDependency.swift [hunk 2]
```diff
@@ -81,7 +81,24 @@ extension Package.Dependency {
...
-
+  
+    /// Adds a remote package dependency given a branch requirement.
+    ///
+    ///    .package(url: "https://example.com/example-package.git", revision: "aa681bd6c61e22df0fd808044a886fc4a7ed3a65"),
+    ///
+    /// - Parameters:
+    ///     - name: The name of the package, or nil to deduce it from the URL.
+    ///     - url: The valid Git URL of the package.
+    ///     - revision: A dependency requirement. See static methods on `Package.Dependency.Requirement` for available options.
```
Comment (abertelrud) [inline Sources/PackageDescription/PackageDependency.swift:92]: This comment as well.

File: Sources/PackageDescription/PackageDependency.swift [hunk 3]
```diff
@@ -81,7 +81,24 @@ extension Package.Dependency {
     ) -> Package.Dependency {
         return .init(name: name, url: url, requirement: .branch(branch))
     }
-
+  
+    /// Adds a remote package dependency given a branch requirement.
```
Comment (abertelrud) [inline Sources/PackageDescription/PackageDependency.swift:85]: Should this say revision?

File: Documentation/PackageDescription.md [hunk 4]
```diff
@@ -306,6 +306,16 @@ static func package(url: String, _ requirement: Package.Dependency.Requirement)
 ///     - branch: A dependency requirement. See static methods on `Package.Dependency.Requirement` for available options.
 static func package(name: String? = nil, url: String, branch: String) -> Package.Dependency
 
+/// Adds a remote package dependency given a branch requirement.
```
Comment (abertelrud) [inline Documentation/PackageDescription.md:309]: Should this say revision?

**Comment by tomerd** [comment]
@swift-ci please smoke test

File: Documentation/PackageDescription.md (hunk 4 above)
Comment (miggs597) [reply Documentation/PackageDescription.md:309]: Yeah you're right, thanks for catching that

**Comment by miggs597** [comment]
@swift-ci please smoke test

File: Sources/PackageDescription/PackageDependency.swift (hunk 3 above)
Comment (tomerd) [reply Sources/PackageDescription/PackageDependency.swift:85]: this still say "branch" ^^

**Comment by miggs597** [comment]
@swift-ci please smoke test

**Review by tomerd — APPROVED** [review]

**Pull request merged**